src/BinaryParsers/PEBinary/PortableExecutable/PE.cs

```diff
@@ -789,5 +789,66 @@ public bool IsWixBinary
                 return this.isWixBinary.Value;
             }
         }
+
+        public bool IsChecksumAlgorithmSecureForPortablePdb()

```

Comment:
>IsChecksumAlgorithmSecureForPortablePdb [](start = 20, length = 39)

rather than creating these helpers, you should add a helper that simply returns the checksum algorithm for managed PDBs. 

ManagedPdbSourceFileChecksumAlgorithm { get; }

Reply:
This property s/be on the PDB class, not in the PE class.

---
In reply to: [563213952](https://github.com/microsoft/binskim/pull/335#discussion_r563213952) [](ancestors = 563213952)

Reply:
oh...remembered why i didn't move the below logic to pdb.cs: to do th reading I need a few things: peReader and the metadatareader.

---
In reply to: [563214137](https://github.com/microsoft/binskim/pull/335#discussion_r563214137) [](ancestors = 563214137,563213952)

Reply:
Changed a little bit, let me know what do u think.

---
In reply to: [563297494](https://github.com/microsoft/binskim/pull/335#discussion_r563297494) [](ancestors = 563297494,563214137,563213952)